views/js/checkout-component-renderer.js

```diff
@@ -492,17 +497,49 @@ jQuery(document).ready(function() {
         }
 
         function handleOnClick(resolve, reject) {
-            const paymentMethodContainer = this.paymentForm.closest('.adyen-payment');
+            const paymentMethodContainer = getFormPaymentMethodContainer(this.paymentForm);
+            const paymentMethodType = paymentMethodContainer.data('local-payment-method');
             // Show message if button is disabled else if not in progress, hide and resolve
             if (prestaShopPlaceOrderButton.prop('disabled') && !isPlaceOrderInProgress()) {
                 showRequiredConditionsInfoMessage(paymentMethodContainer);
-                reject(new Error('Terms of service not agreed'));
+                if (paymentMethodType === 'paypal') {
+                    return false;
+                } else {
+                    reject(new Error('Terms of service not agreed'));
+                }
             } else if (!isPlaceOrderInProgress()) {
                 hideInfoMessage(paymentMethodContainer);
-                resolve();
+                if (paymentMethodType === 'paypal') {
+                    return true;
+                } else {
+                    resolve();
+                }
             }
         }
 
+        function handleOnCancel(state) {
+            hidePopup();
```

Comment:
do we need to hidePopup() ?

I think it can be removed but pls double check it if it's used anywhere

```suggestion
```